tests/test_results.py

```diff
@@ -98,6 +100,20 @@ def _test_results_dict(self, results_dict):
         for key in expected_keys:
             assert key in results_dict.keys(), '{} key not in results dictionary'.format(key)
 
+    #test eval_number 
+    def test_qaoa_result_eval_number(self):
```

Comment:
As a general rule, a viable test for this feature would be:
- for a deterministic optimizer check that the returned `results.optimized['eval_number']` is the same as the eval corresponding to the lowest value in the cost history.

Can we implement it?